Service/NotificationManager.php

```diff
@@ -0,0 +1,170 @@
+<?php
+  /**
+   * Created by groovili
+   * https://github.com/groovili
+   */
+  declare(strict_types = 1);
+
+  namespace Kopay\NotificationBundle\Service;
+
+  use Kopay\NotificationBundle\Entity\NotificationEmail;
+  use Kopay\NotificationBundle\Entity\NotificationMessageInterface;
+  use Kopay\NotificationBundle\Entity\NotificationPush;
+  use Symfony\Component\OptionsResolver\Exception\InvalidArgumentException;
+
+  /**
+   * Class NotificationManager
+   *
+   * @package Kopay\NotificationBundle\Service
+   */
+  class NotificationManager {
+    /**
+     * @var array
+     */
+    private $configuration;
+
+    protected const NOTIFICATION_EMAIL = 0;
+
+    protected const NOTIFICATION_PUSH  = 1;
+
+    /**
+     * @var NotificationMessageInterface
+     */
+    protected $notification = null;
+
+    /**
+     * NotificationManager constructor.
+     *
+     * @param array $config
+     */
+    public function __construct(array $config)
+    {
+      $this->configuration = $config;
+    }
+
+    /**
+     * @param int $type
+     *
+     * @return \Kopay\NotificationBundle\Service\NotificationManager|null
+     */
+    public function create(int $type): ?NotificationManager
+    {
+      if ($type !== self::NOTIFICATION_EMAIL && $type !== self::NOTIFICATION_EMAIL) {
+        throw new InvalidArgumentException('Unknown type of notification.');
+      }
+
+      switch ($type) {
+        case self::NOTIFICATION_EMAIL:
+          $this->notification = new NotificationEmail();
+          break;
+        case self::NOTIFICATION_PUSH:
+          $this->notification = new NotificationPush();
+          break;
+      }
+
+      return $this;
+    }
+
+    /**
+     * @param string $title
+     *
+     * @return \Kopay\NotificationBundle\Service\NotificationManager|null
+     */
+    public function setTitle(string $title): ?NotificationManager
```

Comment:
shouldn't user construct the message himself and then just send it out? Why do we do that for him?